test/Makefile.am

```diff
@@ -0,0 +1,182 @@
+#*******************************************************************************
+#  Copyright (c) 2016
+# 
+#  All rights reserved. This program and the accompanying materials
+#  are made available under the terms of the Eclipse Public License v1.0
+#  and Eclipse Distribution License v1.0 which accompany this distribution. 
+# 
+#  The Eclipse Public License is available at 
+#     http://www.eclipse.org/legal/epl-v10.html
+#  and the Eclipse Distribution License is available at 
+#    http://www.eclipse.org/org/documents/edl-v10.php.
+# 
+#  Contributors:
+#     Guilherme Maciel Ferreira - initial implementation
+#*******************************************************************************/
+
+###############################################################################
+# Binaries names
+###############################################################################
+
+bin_PROGRAMS  =
+
+if PAHO_BUILD_SYNC
+bin_PROGRAMS += test1
```

Comment:
I would use "check_PROGRAMS" instead of "bin_PROGRAMS" here. Because you usually want to run these binaries during "make check". But you do not want to install them.